test/classes-additional.coffee

```diff
@@ -0,0 +1,301 @@
+# Classes
+# -------
+
+# * Class Instantiation
+# * Inheritance and Super
+# * Default parameters
+# * Version Compatability
+test "multiple super calls", ->
+  # class A 
+  #   constructor: (@drink) ->
+  #   make: -> "Making a #{@drink}"
+
+  # class MultiSuper extends A
+  #   constructor: (drink) ->
+  #     super(drink)
+  #     super(drink)
+  #     @newDrink = drink
+  # eq (new MultiSuper('Late')).make(), 'Making a Late'
+
+test "@ params", ->
+  class A 
+    constructor: (@drink, @shots, @flavor) ->
+    make: -> "Making a #{@flavor} #{@drink} with #{@shots} shot(s)"
+
+  a = new A('Machiato', 2, 'chocolate')
+  eq a.make(),  "Making a chocolate Machiato with 2 shot(s)"
+
+  class B extends A
+  b = new B('Machiato', 2, 'chocolate')
+  eq b.make(),  "Making a chocolate Machiato with 2 shot(s)"
+
+test "@ params with defaults", ->
+  class A 
+    # Multiple @ params with defaults
+    constructor: (@drink = 'Americano', @shots = '1', @flavor = 'caramel') ->
+    make: -> "Making a #{@flavor} #{@drink} with #{@shots} shot(s)"
+
+  a = new A()
+  eq a.make(),  "Making a caramel Americano with 1 shot(s)"
+
+test "@ params with class params", ->
+  class Beverage
+    drink: 'Americano'
+    shots: '1'
+    flavor: 'caramel'
+
+  class A 
+    # Class creation as a default param with `this`
+    constructor: (@drink = new Beverage()) ->
+  a = new A()
+  eq a.drink.drink, 'Americano'
+
+  beverage = new Beverage
+  class B 
+    # class costruction with a default external param
+    constructor: (@drink = beverage) ->
+
+  b = new B()
+  eq b.drink.drink, 'Americano'
+
+  class C 
+    # Default constructor with anonymous empty class
+    constructor: (@meta = class) ->
+  c = new C()
+  ok c.meta instanceof Function
+
+test "@ params without super, including errors", ->
+  class A 
+    constructor: (@drink) ->
+    make: -> "Making a #{@drink}"
+  a = new A('Machiato')
+  eq a.make(),  "Making a Machiato"
+
+  throwsB = """
+  class B extends A
+    #implied super
+    constructor: (@drink) ->
+      if @drink is 'Machiato'
+        @cost = 'expensive'
+    make: -> "Making an #{@cost} #{@drink}"
+  b = new B('Machiato', 'Cafe Ole', 'Americano')
+  eq b.make(),  "Making an expensive Machiato" 
+  """
+  throws -> CoffeeScript.run classA + throwsB, bare: yes
+
+  throwsC = """
+  class C extends A
+    # super with splats
+    constructor: (@params...) ->
+      super(@params[0])
+      console.log @params
+
+  c = new C('Machiato', 'Cafe Ole', 'Americano')
+  # eq c.make(),  "Making a Machiato" 
+  """
+  throws -> CoffeeScript.run classA + throwsC, bare: yes
+
+
+test "super and external constructors", ->
+  ctorA = (@drink) ->
+  class A 
+    constructor: ctorA
+    make: -> "Making a #{@drink}"
+
+  # # External constructor with super
+  # ctorB = (drink, flavor) ->
+  #   super(drink)
```

Comment:
So this wouldn't work on `master` either, so I'd either turn this into an error check or remove it.